### ISSUE TITLE
Doesn't seem to work with the keon.

### ISSUE THREAD

**Issue by Floppymnt** [UncensorPat/prohe#3]
As the title states, it doesn't seem to support some toys. What would be necessary to add support?

**Comment by UncensorPat** [comment]
Currently the codebase is only designed to work with vibration-enabled devices, not with stroker devices. I don't think it would be particularly difficult to add support for stroker devices, however I don't have the time to make the change nor any of the devices necessary to test with.

The change necessary is entirely encapsulated in `src/extension.ts`, and should just consist of:

1. Updating `updateDeviceList()` to also add devices with (linear attributes)[https://buttplugio.github.io/buttplug-js/classes/ButtplugClientDevice.html#linearAttributes]
2. Generifying `updateDeviceVibration()` such that `device.vibrate(status.level);` is replaced with a section that does whatever is necessary based on the type of device it is

**Comment by Floppymnt** [comment]
Yeah, I've been trying to do this for a while now and it isn't working. I know it's a problem on my part. I'm just not experienced enough with typescript. I managed to get it connected but anything other than that is beyond me. :/ 

Thanks for the nudge in the right direction though.

**Comment by UncensorPat** [comment]
Reopening as I do want to resolve this at some point, I just don't have the hardware to test any changes I make. I discussed the ability to test with a simulator with qdot in the Discord the other day and they suggested that while nothing exists at the moment, they may look to implement a simulator in the near future. If that gets created then I can look to add this support.

**Comment by Floppymnt** [comment]
I might have been able to figure it out myself if they had documentation I could understand. I'm not sure if it's because it's done poorly or if I'm not smart enough to understand it. On the note of not having a test device, if you want I can test the code on mine.
